src/main/java/org/broadinstitute/hellbender/utils/codecs/FeatureOutputCodec.java

```diff
@@ -5,11 +5,15 @@
 import org.broadinstitute.hellbender.engine.GATKPath;
 
 import java.io.IOException;
+import java.util.Comparator;
 import java.util.List;
+import java.util.PriorityQueue;
 
 public interface FeatureOutputCodec<F extends Feature, S extends FeatureSink<F>> {
     boolean canDecode( String path );
     Class<F> getFeatureType();
     S makeSink( GATKPath path, SAMSequenceDictionary dict, List<String> sampleNames, int compressionLevel );
-    void encode( F feature, S sink  ) throws IOException;
+    void encode( F feature, S sink ) throws IOException;
+    Comparator<F> getSameLocusComparator();
+    void resolveSameLocusFeatures( PriorityQueue<F> queue, S sink );
```

Comment:
Moved resolution of same-locus features into a general post-processing companion class (which would be available for other situations).  One could hook these up as part of a streaming operation, I think, which would allow your suggested use case.
I really wish that features could implement Comparable.  Alas, they require a dictionary to compare.  This could be encapsulated in a Comparable, but then you'd have to pass the comparable around (instead of passing a dictionary around), and so you're no further ahead.